src/core_plugins/input_control_vis/public/control/control.test.js

```diff
@@ -116,3 +117,34 @@ describe('ancestors', () => {
     expect(ancestorFilters[1]).to.eql('mockKbnFilter:myGrandParentValue');
   });
 });
+
+describe('empty state', () => {
+  class SimpleFilterManager extends FilterManager {
+    getValueFromFilterBar() {
+      return this.getUnsetValue();
+    }
+    createFilter() {
+      return {};
+    }
+  }
+  const indexPatternMock = {};
+  const queryFilterMock = {};
+  const controlParams = {};
+  const filterManager = new SimpleFilterManager('controlId', 'myField', indexPatternMock, queryFilterMock, 'originalUnsetValue');
+  const control = new Control(controlParams, filterManager);
+
+  test('should be in empty state when created', function () {
+    expect(control.value).to.be('originalUnsetValue');
+  });
+
+  test('should update control value to new empty state when unset value changes and control is in empty state', function () {
+    control.setUnsetValue('updatedUnsetValue');
+    expect(control.value).to.be('updatedUnsetValue');
+  });
+
+  test('should not update control value when unset value changes and control is has a value', function () {
+    control.set(42);
+    control.setUnsetValue('evenNewerUnsetValue');
+    expect(control.value).to.be(42);
+  });
+});
```

Comment:
This doesn't really cover the case I was thinking of which is something like this:

```
    control.set('originalUnsetValue');
    control.setUnsetValue('evenNewerUnsetValue');
    expect(control.value).to.be(?); // Should this be evenNewerUnsetValue originalUnsetValue? 
```
Your current code would make the final value be 'evenNewerUnsetValue', which maybe makes sense, but it's just a little confusing because the user explicitly set the value to something and then it got changed.

What about this test case:
```
    control.setUnsetValue('value1');
    control.set('value2');
    expect(control.hasValue()).to.be(true);
    control.setUnsetValue('value2');
   expect(control.hasValue()).to.be(false); 
```
  
Actually, another interesting case that's even simpler:
```
    control.setUnsetValue('originalUnsetValue');
    control.set('originalUnsetValue');
    expect(control.hasValue()).to.be(?);  // Should this be true or false? Does the control differentiate between the user explicitly setting a value, vs the default/unset value?
```

Maybe I'm still confused about how this unset value works but if it's like a default value (which I guess it isn't but I'm still not sure how exactly) then a default value of X is different than the user explicitly setting a value of X.  

But adding in some of these tests cases so at least the contract is clear I think would be good.